### PR TITLE
windows: Implement DLL metadata

### PR DESCRIPTION
**Pull request by amyspark**
Hi @dragonmux,

This PR implements metadata for the shared library version of Substrate. Let me know what you think.

### CODE REVIEW

File: impl/substrate.rc.in [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+BEGIN
+	BLOCK "StringFileInfo"
+	BEGIN
+		BLOCK "040904E4"
+		BEGIN
+			VALUE "CompanyName",      "bad_alloc heavy industries"
+			VALUE "FileDescription",  @VER_FILEDESCRIPTION_STR@
+			VALUE "FileVersion",      @VER_FILEVERSION_STR@
+			VALUE "InternalName",     ""
+			VALUE "LegalCopyright",   "Copyright (c) 2020, Rachel Mant, Aki Van Ness. All rights reserved. SPDX-License-Ref: @license@"
```
Comment (dragonmux) [inline impl/substrate.rc.in:19]: We've been pretty bad at keeping these notices updated (oops!) but this should probably be made to read `2020-2023` for the copyright years

File: impl/meson.build [hunk 2]
```diff
@@ -148,6 +148,31 @@ if library_type == 'static' and isWindows
 	libSubstrateArgs = ['-DSUBSTRATE_BUILD_STATIC']
 endif
 
+if isWindows
+	rc_cfg = configuration_data()
```
Comment (dragonmux) [inline impl/meson.build:152]: `rcConfig` or something similar? The point isn't the exact name suggestion, it's that these need lowerCamelCase'ing to match the style of the rest of the file

**Review by dragonmux — CHANGES_REQUESTED** [review]
Spotted a couple of easily fixed oopses, which once rectified, we're happy to merge this

File: impl/substrate.rc.in (hunk 1 above)
Comment (amyspark) [reply impl/substrate.rc.in:19]: Your license information says 2020, hence why I preserved that information.

File: impl/meson.build [hunk 3]
```diff
@@ -148,6 +148,31 @@ if library_type == 'static' and isWindows
...
+if isWindows
+	rcCfg = configuration_data()
+	if cxx.get_id() == 'msvc' or cxx.get_id() == 'clang-cl'
+		rcCfg.set_quoted('VER_ORIGINALFILENAME_STR', 'substrate.dll')
+	else
+		rcCfg.set_quoted('VER_ORIGINALFILENAME_STR', 'libsubstrate.dll')
+	endif
+	rcCfg.set_quoted('VER_FILEDESCRIPTION_STR', 'A collection of headers containing useful tools and gadgets for building C++ programs')
+	rcCfg.set_quoted('VER_PRODUCTNAME_STR', meson.project_name())
+	project_version = '@0@,0'.format(meson.project_version().replace('.', ','))
```
Comment (dragonmux) [inline impl/meson.build:160]: `project_version` still needs lowerCamelCasing please

**Review by dragonmux — COMMENTED** [review]
One outstanding item remains and once that's fixed, this is good to merge - thank you!

File: impl/substrate.rc.in (hunk 1 above)
Comment (dragonmux) [reply impl/substrate.rc.in]: Aye, it needs fixing too

**Review by dragonmux — APPROVED** [review]
LGTM now, merging!

**Pull request merged**